### PR TITLE
Extend searchable snapshots feature usage stats

### PR DESCRIPTION
**Pull request by ywelsch**
Extends the feature usage stats, distinguishing between full_copy and shared_cache style searchable snapshot indices.

### CODE REVIEW

**Review by tlrx — APPROVED** [review]
LGTM

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -112,6 +112,7 @@ testClusters.all {
   setting 'indices.lifecycle.history_index_enabled', 'false'
   keystore 'bootstrap.password', 'x-pack-test-password'
   keystore 'xpack.security.transport.ssl.secure_key_passphrase', 'testnode'
+  setting 'xpack.searchable.snapshot.shared_cache.size', '10mb'
```
Comment (original-brownbear) [inline x-pack/plugin/build.gradle:115]: More of a comment/note: One somewhat unfortunate side-effect of this change is that we now don't have any REST test coverage for the `0b` cache size use case right?
Should we make sure that a node with `0b` cache size continues to properly start up in some form still? (It was quite helpful to have this coverage in my recent work)

**Review by original-brownbear — APPROVED** [review]
LGTM

File: x-pack/plugin/build.gradle (hunk 1 above)
Comment (ywelsch) [reply x-pack/plugin/build.gradle:115]: I'm not sure if that functionality needs a REST test. As we already start up integration tests that way, that might give us good enough coverage?

**Pull request merged**